packages/usa-modal/src/index.js

```diff
@@ -170,104 +171,151 @@ function toggleModal(event) {
+};
 
-  // Rebuild the modal element
-  modalContent.parentNode.insertBefore(modalWrapper, modalContent);
-  modalWrapper.appendChild(modalContent);
-  modalContent.parentNode.insertBefore(overlayDiv, modalContent);
-  overlayDiv.appendChild(modalContent);
-
-  // Add classes and attributes
-  modalWrapper.classList.add(HIDDEN_CLASS);
-  modalWrapper.classList.add(WRAPPER_CLASSNAME);
-  overlayDiv.classList.add(OVERLAY_CLASSNAME);
+/**
+ * Moves necessary attributes from Modal HTML to wrapper element.
+ *
+ * @param {HTMLDivElement} baseComponent - Modal HTML in the DOM.
+ * @param {HTMLDivElement} targetWrapper - Modal component wrapper element.
+ * @returns Modal wrapper with correct attributes.
+ */
+const setModalAttributes = (baseComponent, targetWrapper) => {
+  const modalID = baseComponent.getAttribute("id");
+  const ariaLabelledBy = baseComponent.getAttribute("aria-labelledby");
+  const ariaDescribedBy = baseComponent.getAttribute("aria-describedby");
+  const forceUserAction = baseComponent.hasAttribute(FORCE_ACTION_ATTRIBUTE);
 
-  // Set attributes
-  modalWrapper.setAttribute("role", "dialog");
-  modalWrapper.setAttribute("id", modalID);
+  if (!ariaLabelledBy)
+    throw new Error(`${modalID} is missing aria-labelledby attribute`);
 
-  if (ariaLabelledBy) {
-    modalWrapper.setAttribute("aria-labelledby", ariaLabelledBy);
-  }
+  if (!ariaDescribedBy)
+    throw new Error(`${modalID} is missing aria-desribedby attribute`);
 
-  if (ariaDescribedBy) {
-    modalWrapper.setAttribute("aria-describedby", ariaDescribedBy);
-  }
+  // Set attributes
+  targetWrapper.setAttribute("role", "dialog");
+  targetWrapper.setAttribute("id", modalID);
+  targetWrapper.setAttribute("aria-labelledby", ariaLabelledBy);
+  targetWrapper.setAttribute("aria-describedby", ariaDescribedBy);
 
   if (forceUserAction) {
-    modalWrapper.setAttribute(FORCE_ACTION_ATTRIBUTE, "true");
+    targetWrapper.setAttribute(FORCE_ACTION_ATTRIBUTE, forceUserAction);
   }
 
+  // Add aria-controls
+  const modalClosers = targetWrapper.querySelectorAll(CLOSERS);
+  modalClosers.forEach((el) => {
+    el.setAttribute("aria-controls", modalID);
+  });
+
   // Update the base element HTML
   baseComponent.removeAttribute("id");
   baseComponent.removeAttribute("aria-labelledby");
   baseComponent.removeAttribute("aria-describedby");
   baseComponent.setAttribute("tabindex", "-1");
 
-  // Add aria-controls
-  const modalClosers = modalWrapper.querySelectorAll(CLOSERS);
-  modalClosers.forEach((el) => {
-    el.setAttribute("aria-controls", modalID);
-  });
+  return targetWrapper;
+};
+
+/**
+ * Takes in Modal HTML and builds component w/ correct structure.
+ *
+ * @param {HTMLDivElement} baseComponent - Modal HTML in the DOM.
+ * @returns Modal component - Modal wrapper w/ nested Overlay and Modal Content.
+ */
+const rebuildModal = (baseComponent) => {
+  const modalContent = baseComponent;
+  const modalWrapper = document.createElement("div");
```

Comment:
Optional:
For clarity, it might help to give more context about what the wrapper is wrapping
```suggestion
  const modalContentWrapper = document.createElement("div");
```

Reply:
Resolved in https://github.com/uswds/uswds/pull/5315/commits/c84d5ed04ad0b2cbf9d9dfec339df203d343da9c